src/main/java/net/minestom/server/particle/ParticleCreator.java

```diff
@@ -23,14 +31,27 @@ public static ParticlePacket createParticlePacket(Particle particleType, boolean
         } else {
             data = new byte[0];
         }
+
         return new ParticlePacket(particleType.id(), distance, x, y, z, offsetX, offsetY, offsetZ, particleData, count, data);
     }
 
-    public static ParticlePacket createParticlePacket(Particle particleType,
+
+    public static ParticlePacket createParticlePacket(@NotNull Particle particleType, boolean distance,
+                                                      double x, double y, double z,
+                                                      float offsetX, float offsetY, float offsetZ,
+                                                      float particleData, int count, ParticleOption options) {
+
+        Check.argCondition(particleType != options.type(), "Particle type ''{0}'' does not support Particle options ''{1}''",
+                particleType.namespace(), options.getClass().getSimpleName());
+        return new ParticlePacket(particleType.id(), distance, x, y, z, offsetX, offsetY, offsetZ, particleData, count, options);
+    }
+
+
+    public static ParticlePacket createParticlePacket(Particle particleType, boolean distance,
```

Comment:
This method is bad because you can still cause client crashes by giving Particle.DUST, for example.
Instead, the previous method should be used which ensures that you give options when they are needed.

Reply:
Done